lib/ast/AST.cpp

```diff
@@ -902,6 +902,31 @@ std::map<std::string, int> KORECompositePattern::gatherVarCounts(void) {
   return result;
 }
 
+sptr<KOREPattern> KORECompositePattern::dedupeDisjuncts(void) {
+  if (constructor->getName() != "\\or") {
+    return shared_from_this();
+  }
+  std::vector<sptr<KOREPattern>> items, dedupedItems;
+  flatten(this, "\\or", items);
+  std::set<std::string> printed;
+  for (sptr<KOREPattern> item : items) {
+    std::ostringstream Out;
+    item->print(Out);
+    if (printed.insert(Out.str()).second) {
+      dedupedItems.push_back(item);
+    }
+  }
```

Comment:
It may well be the best way to do it, but I'm curious why we don't have some kind of structural equality on these patterns other than the one induced by printing them.

Reply:
Just was never implemented. This doesn't seem to be particularly performance intensive and it was a convenient way to implement it, so I went ahead with it. I can try to implement an operator== if you'd rather?